src/libutil/compression.cc

```diff
@@ -302,11 +376,11 @@ struct XzSink : CompressionSink
 #ifdef HAVE_LZMA_MT
 struct ParallelXzSink : public XzSink
 {
-  ParallelXzSink(Sink &nextSink) : XzSink(nextSink, [this]() {
+  ParallelXzSink(Sink &nextSink, int level) : XzSink(nextSink, [this](unsigned level) {
```

Comment:
Can you capture `level` in the lambda, and avoid having to declare/pass the parameter in?

Reply:
First--yikes indeed re:this mess.  :(.  Apologies.

That out of the way, I actually originally captured level in the lambda but then the `checkLevel(..)` was either duplicated or a new function....

Hmm I think I just left this to focus on getting things working first--- but never came back to try to do something reasonable here O:).  Good call, thank you.